### ISSUE TITLE
✅ 🐞 Editor History doesn't log individual actions when copy & paste keyboard shortcuts are used - [ED-14553]

### ISSUE THREAD

**Issue by HeinrichMarketing** [elementor/elementor#25861]
### Prerequisites

- [X] I have searched for similar issues in open and closed tickets and cannot find a duplicate.
- [X] I have troubleshooted my issue, and it still exists against the latest stable version of Elementor.

### Description

I'm not sure if it's because of the newest update or what but when I hit Ctrl + Z once to go back once, it undid all of the work. I now can't seem to get it back even when pressing Ctrl + Shift + Z.

### Steps to reproduce

i work with flexbox, new tabs 

### Expected behavior

when pressing ctrl z it should go 1 step back, and not all... 

### Isolating the problem

- [ ] This bug happens when only the Elementor (and Elementor Pro) plugins are active.
- [ ] This bug happens with the Hello Elementor theme active.
- [X] I can reproduce this bug consistently by following the steps I described above.

### Elementor System Info

<details>
<summary>Click to reveal</summary>

````txt

					
== Server Environment ==
	Operating System: Linux
	Software: LiteSpeed
	MySQL version: MariaDB Server v10.5.24
	PHP Version: 8.1.28
	PHP Memory Limit: 2048M
	PHP Max Input Vars: 1000
	PHP Max Post Size: 32M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.5.2
	Site URL: https://schaedler-recht.de
	Home URL: https://schaedler-recht.de
	WP Multisite: No
	Max Upload Size: 32 MB
	Memory limit: 40M
	Max Memory limit: 2048M
	Permalink Structure: /%postname%/
	Language: de_DE
	Timezone: Europe/Berlin
	Admin Email: webmaster@heinrich-marketing.de
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor
	Version: 3.0.1
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: de-DE
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36

== Active Plugins ==
	Borlabs Cookie
		Version: 3.0.6
		Author: Borlabs GmbH

	Elementor
		Version: 3.21.3
		Author: Elementor.com

	Elementor Pro
		Version: 3.21.1
		Author: Elementor.com

	eRecht24 legal texts for WordPress
		Version: 3.3.11
		Author: eRecht24, Weslink GmbH

	Happy Elementor Addons
		Version: 3.10.7
		Author: Leevio

	Happy Elementor Addons Pro
		Version: 2.11.3
		Author: Leevio

	LiteSpeed Cache
		Version: 6.1
		Author: LiteSpeed Technologies

	SEOPress
		Version: 7.6.1
		Author: The SEO Guys at SEOPress

	SEOPress PRO
		Version: 7.6.1
		Author: The SEO Guys at SEOPress

	ShortPixel Image Optimizer
		Version: 5.5.5
		Author: ShortPixel - Convert WebP/AVIF & Optimize Images


== Elemente Verwendung ==
	
	container : 0
		button : 1
		icon-list : 2
		image : 2
		spacer : 1
		text-editor : 2
	section : 2
		button : 2
		divider : 2
		heading : 3
		spacer : 2
		text-editor : 3
	wp-page : 43
		button : 40
		container : 153
		divider : 38
		erecht24 : 2
		google_maps : 1
		ha-advanced-tabs : 1
		heading : 195
		html : 1
		icon : 1
		icon-list : 11
		image : 42
		nested-tabs : 3
		spacer : 15
		text-editor : 155


== Einstellungen ==
	
	cpt_support: post, page
	disable_color_schemes: yes
	disable_typography_schemes: yes
	allow_tracking: yes
	unfiltered_files_upload: 1
	font_display: swap
	load_fa4_shim: yes
	disabled_elements: video, counter, progress, testimonial, social-icons, alert, audio, shortcode, sidebar, read-more, rating, wp-widget-pages, wp-widget-calendar, wp-widget-archives, wp-widget-media_audio, wp-widget-media_image, wp-widget-media_gallery, wp-widget-media_video, wp-widget-meta, wp-widget-search, wp-widget-text, wp-widget-categories, wp-widget-recent-posts, wp-widget-recent-comments, wp-widget-rss, wp-widget-tag_cloud, wp-widget-nav_menu, wp-widget-custom_html, wp-widget-block, wp-widget-seopress_pro_lb_widget, sp-breadcrumbs, text-path, theme-site-logo, theme-site-title, theme-page-title, theme-post-excerpt, theme-post-featured-image, theme-archive-title, archive-posts, portfolio, login, slides, hotspot, price-list, price-table, flip-box, call-to-action, media-carousel, reviews, table-of-contents, countdown, share-buttons, search-form, author-box, post-comments, post-navigation, blockquote, facebook-button, facebook-comments, facebook-embed, facebook-page, code-highlight, video-playlist, paypal-button, stripe-button, progress-tracker, menu-anchor


== Eigenschaften ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	
	recaptcha_v3: Aktiv


== Elementor-Experimente ==
	Verbessertes Laden von Assets: Standardmäßig aktiviert
	Verbessertes Laden von CSS: Standardmäßig aktiviert
	Inline-Schriftarten-Symbole: Standardmäßig deaktiviert
	Additional Custom Breakpoints: Standardmäßig aktiviert
	admin_menu_rearrangement: Standardmäßig deaktiviert
	Flexbox Container: Aktiv
	Upgrade Swiper Library: Aktiv
	Grid Container: Aktiv
	Nested Elements Performance: Standardmäßig deaktiviert
	Optimized Control Loading: Standardmäßig deaktiviert
	Default to New Theme Builder: Standardmäßig aktiviert
	Hello Theme Header & Footer: Standardmäßig aktiviert
	Elementor Home Screen: Standardmäßig aktiviert
	Editor Top Bar: Standardmäßig deaktiviert
	Build with AI: Standardmäßig aktiviert
	Startseiten: Inaktiv
	Nested Elements: Aktiv
	Hintergrundbilder nachladen: Aktiv
	Bedingungen anzeigen: Standardmäßig deaktiviert
	Form Submissions: Inaktiv
	Menü: Standardmäßig deaktiviert
	Taxonomy Filter: Standardmäßig deaktiviert


== Protokoll ==
	
Log: showing 20 of 332024-04-11 18:20:55 [info] elementor-pro::elementor_pro_updater Started 
2024-04-11 18:20:55 [info] Elementor Pro/Upgrades - _on_each_version Start  
2024-04-11 18:20:55 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2024-04-11 18:20:55 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.19.3',
  'to' => '3.20.3',
)]
2024-04-11 18:20:55 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.19.3',
  'to' => '3.20.3',
)]
2024-04-11 18:21:21 [info] elementor::elementor_updater Started 
2024-04-11 18:21:21 [info] Elementor/Upgrades - _on_each_version Start  
2024-04-11 18:21:21 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.19.4',
  'to' => '3.20.4',
)]
2024-04-11 18:21:21 [info] Elementor/Upgrades - _on_each_version Finished 
2024-04-11 18:21:21 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.19.4',
  'to' => '3.20.4',
)]
2024-04-11 18:21:22 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.19.4',
  'to' => '3.20.4',
)]
2024-04-26 12:33:49 [info] elementor-pro::elementor_pro_updater Started 
2024-04-26 12:33:49 [info] Elementor Pro/Upgrades - _on_each_version Start  
2024-04-26 12:33:49 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2024-04-26 12:33:49 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.20.3',
  'to' => '3.21.1',
)]
2024-04-26 12:33:49 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.20.3',
  'to' => '3.21.1',
)]
2024-04-26 12:34:42 [info] elementor::elementor_updater Started 
2024-04-26 12:34:43 [info] Elementor/Upgrades - _on_each_version Start  
2024-04-26 12:34:43 [info] Elementor/Upgrades - _on_each_version Finished 
2024-04-26 12:34:43 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.20.4',
  'to' => '3.21.3',
)]

JS: showing 6 of 6JS: 2024-02-08 09:39:34 [error X 4]../wp-content/plugins/elementor-pro/assets/js/editor.min.js?ver=3.18.1:2:64227] Cannot convert undefined or null to object 
JS: 2024-02-08 11:31:59 [error X 68]../wp-content/plugins/elementor-pro/assets/lib/sticky/jquery.sticky.min.js?ver=3.18.1:1:1162] Failed to execute 'getComputedStyle' on 'Window': parameter 1 is not of type 'Element'. 
JS: 2024-04-12 07:53:12 [error X 5]../wp-content/plugins/elementor/assets/js/ai.min.js?ver=3.20.4:3:7564] ReactUtils is not defined 
JS: 2024-04-26 09:30:39 [error X 1]../wp-content/plugins/elementor-pro/assets/js/frontend.min.js?ver=3.20.3:2:12628] Failed to execute 'observe' on 'IntersectionObserver': parameter 1 is not of type 'Element'. 
JS: 2024-04-26 09:58:56 [error X 1]../wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.21.3:3:654174] Cannot read properties of undefined (reading 'isDesignable') 
JS: 2024-04-26 10:21:50 [error X 6]../wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.21.3:3:821594] Cannot read properties of undefined (reading 'localized') 

PHP: showing 4 of 4PHP: 2024-02-20 17:22:53 [warning X 187][../wp-content/plugins/elementor-pro/modules/forms/classes/recaptcha-v3-handler.php::124] Undefined array key "score" [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-02-22 16:41:26 [warning X 3][../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php::150] Undefined array key "condition_type" [array (
  'trace' => '
#0: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(150): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_template_json_item()
#2: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(120): class type array_map()
#3: ../wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(59): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_templates_json()
#4: ../wp-content/plugins/elementor/data/base/endpoint.php(158): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> get_items()
',
)]
PHP: 2024-04-11 19:09:39 [warning X 2][../wp-content/plugins/elementor/includes/managers/image.php::113] Trying to access array offset on value of type bool [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-04-26 10:19:23 [warning X 4][../wp-content/plugins/elementor/modules/ai/connect/ai.php::99] Array to string conversion [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Kompatibel
	Happy Elementor Addons: Kompatibel
	Happy Elementor Addons Pro: Kompatibel

== Elementor Pro - Compatibility Tag ==
	
	Happy Elementor Addons: Kompatibel
	Happy Elementor Addons Pro: Kompatibel

````

</details>

### Agreement

- [X] I agree that my issue may be closed without action if it doesn't meet all the requirements.

**Comment by gorfreed** [comment]
Same here, I am experiencing issues with using revisions or simply "undo" across the board on our projects. It seems to be related to the latest updates and is very worrying as a lot of editing work has been lost because of this.

**Comment by brent-e** [comment]
The same issue for me, since updating to the latest version of Pro. It's intermittent and I can't detect any particular pattern.

**Comment by maximedesro** [comment]
I also noticed issues recently with CTRL Z, and it used to work flawlessly. 

Since the most recent version (3.2.x) any changes made in the HTML element won't event show up in the History.

**Comment by vgstef** [comment]
Same here. It's very worrying, because I was used to use Ctrl-Z. But now, all changes since lasted save (or automatic save?) is lost. I've lost hours of work on different pages because of this. :(

**Comment by gorfreed** [comment]
This has gotten so bad, I restored to 3.19 where this issue does not occur.

**Comment by gorfreed** [comment]
Also I wanted to add, that the Elementor builder took an insane amount of resources from the server, making everything super sluggish and the slower it got, the more likely it was for revisions/CTRL+Z to malfunction.

**Comment by dreamscape-mkt** [comment]
Just leaving a comment here how we replicated it as well. When you hit ctrl + c and then ctrl + v on a widget using the keyboard command it only stores the history of the first copy and paste of the widget but anything after that no history is saved. So when you hit ctrl + z it goes back to the first copy and paste we did with the keyboard command. Right clicking and duping the widget works fine but using the keyboard commands seem to break histories.

**Comment by Henzxon** [comment]
having the same issue as well.

**Comment by mikeKstm** [comment]
Same here. This is really anoying. 😒

**Comment by tamelander** [comment]
Me too.

**Comment by KingGrizzly** [comment]
Same. Started with 3.21.0 for me. Mentioned it here 1.5 weeks ago. https://www.facebook.com/groups/Elementors/permalink/7139246716201053/

**Comment by gbeadv** [comment]
Same here

**Comment by SvenEbens** [comment]
Same here thought I was going cazy, but experienced it on a lot of different sites using elementor.  Can get stuff back through the history browser or with crtl-Y, but takes forever to load history and it's annoying as using ctrl-c/v and ctrl-z is something I have used forever, so this happens a lot now. Hopefully this can be fixed soon.

**Comment by Kraboobee** [comment]
Same here. Revision history is different from what I actually saved, and hitting redo redoes a random combination of changes I made in between. Lost about 2 days of work.

**Comment by seancswanson** [comment]
We are having this issue as well. Has there been any official word from the Elementor on a fix?

**Comment by nicholaszein** [comment]
Hello everyone!

We are aware of this issue and are already working on a fix! 💪 

As soon as I have more information (or we have a PR ready), I'll update this issue! 😉 

Thank you all for sharing your feedback! 🙏

**Comment by iluvios** [comment]
I used to trust the auto save but to hell with it.
I have lost a coupe of hours a day of work because Im jsut used to do CTRL+Z and it just destroys all the work.

VERY MUCH ANNOYING

This needs to be fixed ASAP

**Comment by alexeygoncharov** [comment]
Get your shift back..I'm leaving to Breakdance

**Comment by daruev** [comment]
+ SAME problem
maximum ANNOYING
I want to destroy my PC after this happened again ANOTHER time, not the first time, 
it’s very ridiculous that such a big product has such stupid errors

**Comment by nicholaszein** [comment]
Update: The fix is being reviewed. A PR is almost ready to be merged.

----

A word of advice: we will not tolerate disrespectful comments or any comment that breaks our [Code of Conduct](https://elemn.to/gh-code-of-conduct). Keep the conversation civilized please.

Thank you

**Comment by nicholaszein** [comment]
Hey everyone!

The review process has finished, and it looks like everything is working as expected again. We are ready to merge.

A patch will be released soon. 🙌

Best regards 💪

**Comment by nicholaszein** [comment]
Hi everyone!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Core v3.21.5! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
